### PR TITLE
Refactor how Pinot catalog config is defined in presto helm chart

### PR DESCRIPTION
**Pull request by harold-kfuse**
## Description
A catalogs map is introduced in the values.yaml, which specifies
the Pinot catalog config in the config map template.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

**Review by xiangfu0 — APPROVED** [review]

File: kubernetes/helm/presto/templates/coordinator/configmap.yaml [hunk 1]
```diff
@@ -30,6 +30,7 @@ data:
     pinot.use-date-trunc={{ .Values.pinot.useDateTrunc }}
     pinot.infer-date-type-in-schema={{ .Values.pinot.inferDateTypeInSchema }}
     pinot.infer-timestamp-type-in-schema={{ .Values.pinot.inferTimestampTypeInSchema }}
+    pinot.use-streaming-for-segment-queries={{ .Values.pinot.useStreamingForSegmentQueries }}
```
Comment (xiangfu0) [inline kubernetes/helm/presto/templates/coordinator/configmap.yaml:33]: You can remove entire `pinot.properties`
then add:
```
{{- range $path, $config := .Values.extraConfigs }}
  {{ $path }}: |
{{- tpl $config $ | nindent 4 -}}
{{- end -}}
```

File: kubernetes/helm/presto/values.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ pinot:
   useDateTrunc: true
   inferDateTypeInSchema: true
   inferTimestampTypeInSchema: true
+  useStreamingForSegmentQueries: false
```
Comment (xiangfu0) [inline kubernetes/helm/presto/values.yaml:35]: in values.yaml, you can directly set config files:
```
catalogs:
    pinot.properties: |-
        connector.name=pinot
        pinot.controller-urls=pinot-controller:9000
        pinot.controller-rest-service=pinot-controller:9000
        pinot.allow-multiple-aggregations=true
        pinot.use-date-trunc=true
        pinot.infer-date-type-in-schema=true
        pinot.infer-timestamp-type-in-schema=true
        pinot.use-streaming-for-segment-queries=true
```

File: kubernetes/helm/presto/templates/coordinator/configmap.yaml (hunk 1 above)
Comment (harold-kfuse) [reply kubernetes/helm/presto/templates/coordinator/configmap.yaml]: Done

File: kubernetes/helm/presto/values.yaml (hunk 2 above)
Comment (harold-kfuse) [reply kubernetes/helm/presto/values.yaml]: Done

**Review by xiangfu0 — APPROVED** [review]
Please ensure you tested this and check the indent is correct :P

**Comment by harold-kfuse** [comment]
> Please ensure you tested this and check the indent is correct :P

Dry-run gives this:

apiVersion: v1
kind: ConfigMap
metadata:
  name: presto-catalog
data:
  pinot.properties: |
    connector.name=pinot
    pinot.controller-urls=pinot-controller:9000
    pinot.controller-rest-service=pinot-controller:9000
    pinot.allow-multiple-aggregations=true
    pinot.use-date-trunc=true
    pinot.infer-date-type-in-schema=true
    pinot.infer-timestamp-type-in-schema=true
    pinot.use-streaming-for-segment-queries=false

**Review by xiangfu0 — APPROVED** [review]

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**